docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md

```diff
@@ -0,0 +1,594 @@
+- **DELETE** - `/api/Category` - Delete a category
+
+- **GET**    - `/api/Comment`  -  Retrieve all the stored comments
+- **POST**   - `/api/Comment`  -  Add new comment
+
+## Before You Begin
+
+- Make sure you have [Python3.x](https://www.python.org) installed on your system.
+
+- We'll use [PostgreSQL](https://www.postgresql.org) database for this tutorial. Install Postgres database if you don't have it installed already.
+
+- Make sure [virtualenv](http://docs.python-guide.org/en/latest/dev/virtualenvs/#lower-level-virtualenv) is installed on your system.
+
+
+## Step 0: Setting up the application
+
+- First, create the structure of the app at any location on your system:
+
+```
+project/
+├── app.py
+├── migrate.py
+├── Model.py
+├── requirements.txt
+├── resources
+│   └── Hello.py
+└── run.py
+```
+
+- Then open up your terminal and CD(change directory) into the app root folder - `project`.
+
+- Now, create and activate a [virtual environment](http://docs.python-guide.org/en/latest/dev/virtualenvs/#lower-level-virtualenv).
+
+> :light: virtualenv is a tool to create isolated Python environments. virtualenv creates a folder which contains all the necessary executables to use the packages that a Python project would need.
+
+Creates virtual environment
+
+```
+python3.6 -m venv env
+```
+
+Activates the virtual environment
+
+```
+source env/bin/activate
+```
+
+Add the folowing extensions to `requirements.txt`:
+
+```
+flask==0.12.2
+flask_restful==0.3.6
+flask_script==2.0.6
+flask_migrate==2.1.1
+marshmallow==2.14.0
+flask_sqlalchemy==2.3.2
+flask_marshmallow==0.8.0
+marshmallow-sqlalchemy==0.13.2
+psycopg2==2.7.3.2
+```
+
+The above file contains all python extensions the API will use.
+
+- [flask](http://flask.pocoo.org) - This is a microframework for Python
+
+- [flask_restful](https://flask-restful.readthedocs.io/) - This is an extension for Flask that adds support for quickly building of REST APIs.
+
+- [flask_script](https://flask-script.readthedocs.io/) - This is an extension that provides support for writing external scripts in Flask.
+
+- [flask_migrate](https://flask-migrate.readthedocs.io) - This is an extension that handles SQLAlchemy database migrations for Flask applications using Alembic.
+
+- [marshmallow](https://marshmallow.readthedocs.io/) - This is an ORM/ODM/framework-agnostic library for converting complex datatypes, such as objects, to and from native Python datatypes. We'll use this for validation. This is used to Serializing and Deserializing Objects.
+
+- [flask_sqlalchemy](http://flask-sqlalchemy.pocoo.org) - This is an extension for Flask that adds support for SQLAlchemy.
+
+- [flask_marshmallow](https://flask-marshmallow.readthedocs.io) - This is an ntegration layer for Flask and marshmallow (an object serialization/deserialization library) that adds additional features to marshmallow.
+
+- [marshmallow-sqlalchemy](https://marshmallow-sqlalchemy.readthedocs.io) - This adds additional features to marshmallow.
```

Comment:
"SQLAlchemy integration with the marshmallow (de)serialization library."

Source: https://marshmallow-sqlalchemy.readthedocs.io/en/latest/